Src/java/elm-fhir/src/test/java/org/cqframework/cql/elm/requirements/fhir/DataRequirementsProcessorTest.java

```diff
@@ -893,6 +900,53 @@ public void TestDataRequirementsAnalysisCase2a() throws IOException {
         //outputModuleDefinitionLibrary(moduleDefinitionLibrary);
     }
 
+    @Test
+    public void TestDataRequirementsAnalysisCase2g() throws IOException {
+        CqlTranslatorOptions translatorOptions = getTranslatorOptions();
+        CqlTranslator translator = setupDataRequirementsAnalysis("TestCases/TestCase2g.cql", translatorOptions);
+        org.hl7.fhir.r5.model.Library moduleDefinitionLibrary = getModuleDefinitionLibrary(translator, translatorOptions, new HashMap<String, Object>());
+
+        /*
+        2g - Equal to a compile-time literal function
+        DataRequirement
+        type: Condition
+        dateFilter: { path: onset, value: Today() }
+
+        define DateTimeEqualToFunction:
+          [Condition] C
+            where C.onset as dateTime = Today()
+        */
+
+        ExpressionDef ed = translator.getTranslatedLibrary().resolveExpressionRef("DateTimeEqualToFunction");
+        assertTrue(ed.getExpression() instanceof Query);
+        Query q = (Query)ed.getExpression();
+        assertTrue(q.getSource() != null && q.getSource().size() == 1);
+        AliasedQuerySource source = q.getSource().get(0);
+        assertTrue(source.getExpression() instanceof Retrieve);
+        Retrieve r = (Retrieve)source.getExpression();
+        assertTrue(r.getDateFilter() != null && r.getDateFilter().size() == 1);
+        DateFilterElement dfe = r.getDateFilter().get(0);
+        assertEquals(dfe.getProperty(), "onset");
+        assertTrue(dfe.getValue() instanceof Interval);
+
```

Comment:
Just for my learning, don't how translator knew DateFilterElement is Interval rather than other potential types.

Reply:
DateFilter values are always represented as an interval. (i.e. if the comparison being captured is to a single point, the point is represented as a unit interval on that point. It makes the downstream processing a bit easier if it can always know that it's dealing with an interval.